File: interviewbit-arrays-min-steps-in-infinite-grid.py
class Solution:
    # @param X : list of integers
    # @param Y : list of integers
    # Points are represented by (X[i], Y[i])
    # @return an integer
    def coverPoints(self, X, Y):
        steps = 0
        row, col = X[0], Y[0]
        for i in range(1, len(X)):
            steps += max(abs(row - X[i]), abs(col - Y[i]))
            row, col = X[i], Y[i]

        return steps

File: interviewbit-arrays-min-steps-in-infinite-grid.java
public class Solution {
    // X and Y co-ordinates of the points in order.
    // Each point is represented by (X.get(i), Y.get(i))
    public int coverPoints(ArrayList<Integer> X, ArrayList<Integer> Y) {
        int steps = 0;
        int currX = X.get(0);
        int currY = Y.get(0);
        for (int i = 1; i < X.size(); i++) {
            steps += Math.max(Math.abs(X.get(i) - currX), Math.abs(Y.get(i) - currY));
            currX = X.get(i);
            currY = Y.get(i);
        }
        return steps;
    }
}
